test/e2e/framework/util.go

```diff
@@ -901,13 +902,23 @@ func RestclientConfig(kubeContext string) (*clientcmdapi.Config, error) {
 // ClientConfigGetter is a func that returns getter to return a config.
 type ClientConfigGetter func() (*restclient.Config, error)
 
-// LoadConfig returns a config for a rest client.
-func LoadConfig() (*restclient.Config, error) {
+// LoadConfig returns a config for a rest client with the UserAgent set to include the current test name.
+func LoadConfig() (config *restclient.Config, err error) {
+	defer func() {
+		if err == nil && config != nil {
```

Comment:
Given that the suite can run in parallel, shouldn't we lock on the defers b/c there could be a race on setting the config object.

Reply:
each invocation of LoadConfig returns a fresh object

Reply:
/lgtm